test/units/utils/test_encrypt.py

```diff
@@ -249,3 +250,55 @@ def test_passlib_bcrypt_salt(recwarn):
 
     result = passlib_obj.hash(secret, salt=repaired_salt, ident=ident)
     assert result == expected
+
+
+def test_do_encrypt_no_lib(mocker: MockerFixture) -> None:
+    """Test AnsibleError is raised when no encryption library is installed."""
+    mocker.patch('ansible.utils.encrypt.HAS_CRYPT', False)
+    mocker.patch('ansible.utils.encrypt.PASSLIB_AVAILABLE', False)
+    with pytest.raises(AnsibleError, match=r"Unable to encrypt nor hash, either libxcrypt \(recommended\), crypt, or passlib must be installed\."):
```

Comment:
Having a blank line before the `with` statement (here and elsewhere) would make this more readable:

```suggestion

    with pytest.raises(AnsibleError, match=r"Unable to encrypt nor hash, either libxcrypt \(recommended\), crypt, or passlib must be installed\."):
```